lib/badge-data.js

```diff
@@ -69,7 +69,8 @@ function makeLogo(defaultNamedLogo, overrides) {
 
   if (isDataUri(maybeDataUri)) {
     // +'s are replaced with spaces when used in query params, this returns them to +'s - #1527
-    return maybeDataUri.replace(/\s/g, '+');
+    // remove all whitespace characters (newlines tabs etc) - #1540
+    return decodeURI(maybeDataUri).replace(/ /g, '+').replace(/\s/g, '');
```

Comment:
Logically, it seems we should perform this transform before invoking `isDataUri`. Thoughts?

Reply:
Looks like it is being decoded beforehand,
The problem seems to be that any string with a newline fails the `isDataUri()` test.

To support multiline strings I think the regex would have to be updated to something like:
```regex
/^data:[^;]+;[^,]+,(\s|.)+$/
```